### ISSUE TITLE
tags are empty on ec2_vpc_route_table creation

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29257]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ec2_vpc_route_table

##### ANSIBLE VERSION
`2.3`

##### SUMMARY
From @jimrazmus on 2016-01-15T22:59:46Z
On the latest devel branch, I consistently see an empty tags field in a new route table's registered variable.  e.g.

changed: [localhost] => {"changed": true, "invocation": {"module_args": {"_ansible_check_mode": false, "_ansible_debug": false, "_ansible_diff": false, "_ansible_no_log": false, "_ansible_verbosity": 3, "aws_access_key": null, "aws_secret_key": null, "ec2_url": null, "lookup": "tag", "profile": null, "propagating_vgw_ids": null, "region": "us-east-1", "route_table_id": null, "routes": [{"destination_cidr_block": "0.0.0.0/0", "gateway_id": "igw-abe225cf"}], "security_token": null, "state": "present", "subnets": ["10.20.1.0/24", "10.20.2.0/24", "10.20.3.0/24", "10.20.4.0/24"], **"tags": {"Name": "dev_public_rtbl", "environment": "dev"}**, "validate_certs": true, "vpc_id": "vpc-c88af7ac"}, "module_name": "ec2_vpc_route_table"}, "route_table": {"id": "rtb-ca291bae", "routes": [{"destination_cidr_block": "10.20.0.0/16", "gateway_id": "local", "instance_id": null, "interface_id": null, "state": "active", "vpc_peering_connection_id": null}], **"tags": {}**, "vpc_id": "vpc-c88af7ac"}}

On the second and subsequent runs, the tags field is populated.  e.g.

ok: [localhost] => {"changed": false, "invocation": {"module_args": {"_ansible_check_mode": false, "_ansible_debug": false, "_ansible_diff": false, "_ansible_no_log": false, "_ansible_verbosity": 3, "aws_access_key": null, "aws_secret_key": null, "ec2_url": null, "lookup": "tag", "profile": null, "propagating_vgw_ids": null, "region": "us-east-1", "route_table_id": null, "routes": [{"destination_cidr_block": "0.0.0.0/0", "gateway_id": "igw-abe225cf"}], "security_token": null, "state": "present", "subnets": ["10.20.1.0/24", "10.20.2.0/24", "10.20.3.0/24", "10.20.4.0/24"], **"tags": {"Name": "dev_public_rtbl", "environment": "dev"}**, "validate_certs": true, "vpc_id": "vpc-c88af7ac"}, "module_name": "ec2_vpc_route_table"}, "route_table": {"id": "rtb-ca291bae", "routes": [{"destination_cidr_block": "10.20.0.0/16", "gateway_id": "local", "instance_id": null, "interface_id": null, "state": "active", "vpc_peering_connection_id": null}, {"destination_cidr_block": "0.0.0.0/0", "gateway_id": "igw-abe225cf", "instance_id": null, "interface_id": null, "state": "active", "vpc_peering_connection_id": null}], **"tags": {"Name": "dev_public_rtbl", "environment": "dev"}**, "vpc_id": "vpc-c88af7ac"}}

Copied from original issue: ansible/ansible-modules-extras#1506

**Comment by ansibot** [comment]
From @ansibot on 2016-01-15T22:59:46Z



@erydo, @wimnat, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (102/100 (102%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
------ The body of the message is 7871 characters long; only the first
------ 5000 or so are included here.
Received: from github-smtp2-ext6.iad.github.net ([192.30.252.197]:35303 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drBCG-001mVV-Qa
	for greg@biomassiv.es; Sun, 10 Sep 2017 17:59:24 -0500
Date: Sun, 10 Sep 2017 15:58:44 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505084324;
	bh=o5hDWDHKDBgrpHcVFSe6aLZBUmGLAfGXBd6JIwHS5MA=;
	h=From:Reply-To:To:Cc:Subject:List-ID:List-Archive:List-Post:
	 List-Unsubscribe:From;
	b=LBDdcfLDoC766Mpfz6JlxZrNAL6QJeUMA7HatcSIhXTr2SNmAH8EZh1wzddUHDyCe
	 CMW6aY2gfof0I/JWL3UkoFNdDX3bxbBdXf/I/HcQTVV5v+XRiRx9cIV7a4dGIij2xT
	 z+tRP9ssy/nn2wXfrObrdZMUEKXt3ylWrkVw/5SQ=

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (109/100 (109%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext1.iad.github.net ([192.30.252.192]:42668 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drBCR-001mac-9Y
	for greg@biomassiv.es; Sun, 10 Sep 2017 17:59:35 -0500
Date: Sun, 10 Sep 2017 15:58:28 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505084309;
	bh=zPAp3EmQkPGQEazl9Y0pDBB9eP82/tAw5Uee0dztMS0=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=KL9xuwwa11jS3ZvgMwLR27VzoKmrVk1eI0pja6ceye17NdEHVcJsosormjKMRp7/E
	 yIxQwclJtrRniqlA+NVrt9f2ZJw0R9DkUq1CEGK5FmnjBdgo8WLU3KIkNZPxJZfLJS
	 47ukmzfY/4qf4cAAWFHzFrOdyeetBFrk7AWldkRY=

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by s-hertel** [comment]
Fixed for 2.4 in dab0e75ce0a1cf02443818dbf0a0ae6478302dbb. I have opened a PR to backport the fix to 2.3: https://github.com/ansible/ansible/pull/29989

**Comment by ansibot** [comment]
cc @willthames
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @jillr @tremble
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py -> https://galaxy.ansible.com/community/aws

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
